binder/src/main/java/io/grpc/binder/internal/BinderTransportSecurity.java

```diff
@@ -78,30 +85,66 @@ public static void attachAuthAttrs(
    * Authentication state is fetched from the call attributes, inherited from the transport.
    */
   private static final class ServerAuthInterceptor implements ServerInterceptor {
+
+    private final ObjectPool<? extends Executor> executorPool;
+
+    ServerAuthInterceptor(ObjectPool<? extends Executor> executorPool) {
+      this.executorPool = executorPool;
+    }
+
+    /**
+     * @param authStatusFuture a Future that is known to be complete, for which it's safe to call
+     *                         {@link ListenableFuture#get()} without blocking the current thread.
+     */
+    private <ReqT, RespT> ServerCall.Listener<ReqT> shortCircuitAuthFuture(
+        ListenableFuture<Status> authStatusFuture,
+        ServerCall<ReqT, RespT> call,
+        Metadata headers,
+        ServerCallHandler<ReqT, RespT> next) {
+      Status authStatus;
+      try {
+        authStatus = Uninterruptibles.getUninterruptibly(authStatusFuture);
+      } catch (ExecutionException e) {
+        authStatus = Status.INTERNAL.withCause(e);
+      } catch (CancellationException e) {
+        authStatus = Status.CANCELLED.withCause(e);
```

Comment:
Not here, because we know the future is already done and we're in the short-circuit function. Removed.

On real futures with async logic, the Futures will come from client-supplied logic, so anything goes. But that's another codepath.